arviz/data/io_pyjags.py

```diff
@@ -0,0 +1,320 @@
+    variables: tp.Optional[tp.Union[str, tp.Iterable[str]]] = None,
+    warmup: bool = False,
+    warmup_iterations: int = 0,
+) -> tp.Tuple[tp.Dict[str, np.ndarray], tp.Dict[str, np.ndarray]]:
+    """
+    Convert PyJAGS samples dictionary to ArviZ format and split warmup samples.
+
+    Parameters
+    ----------
+    pyjags_samples: a dictionary mapping variable names to NumPy arrays of MCMC
+                    chains of samples with shape
+                    (parameter_dimension, chain_length, number_of_chains)
+
+    variables: the variables to extract from the samples dictionary
+    warmup: whether or not to return warmup draws in data_warmup
+    warmup_iterations: the number of warmup iterations if any
+
+    Returns
+    -------
+    A tuple of two samples dictionaries in ArviZ format
+    """
+    data_warmup = OrderedDict()
+
+    if variables is None:
+        variables = list(pyjags_samples.keys())
+    elif isinstance(variables, str):
+        variables = [variables]
+
+    if not isinstance(variables, Sequence):
+        raise TypeError("variables must be of type Sequence or str")
+
+    variables = tuple(variables)
+
+    if warmup_iterations > 0:
+        (warmup_samples, actual_samples,) = _split_pyjags_samples_in_warmup_and_actual_samples(
+            pyjags_samples=pyjags_samples,
+            warmup_iterations=warmup_iterations,
+            variable_names=variables,
+        )
+
+        data = _convert_pyjags_samples_dictionary_to_arviz_samples_dictionary(
+            samples=actual_samples, variable_names=variables
+        )
+
+        if warmup:
+            data_warmup = _convert_pyjags_samples_dictionary_to_arviz_samples_dictionary(
+                samples=warmup_samples, variable_names=variables
+            )
+    else:
+        data = _convert_pyjags_samples_dictionary_to_arviz_samples_dictionary(
+            samples=pyjags_samples, variable_names=variables
+        )
+
+    return data, data_warmup
+
+
+def _split_pyjags_samples_in_warmup_and_actual_samples(
+    pyjags_samples: tp.Dict[str, np.ndarray],
+    warmup_iterations: int,
+    variable_names: tp.Optional[tp.Tuple[str, ...]] = None,
+) -> tp.Tuple[tp.Dict[str, np.ndarray], tp.Dict[str, np.ndarray]]:
+    """
+    Split a PyJAGS samples dictionary into actual samples and warmup samples.
+
+    Parameters
+    ----------
+    pyjags_samples: a dictionary mapping variable names to NumPy arrays of MCMC
+                    chains of samples with shape
+                    (parameter_dimension, chain_length, number_of_chains)
+
+    warmup_iterations: the number of draws to be split off for warmum
+    variable_names: the variables in the dictionary to use; if None use all
+
+    Returns
+    -------
+    A tuple of two pyjags samples dictionaries in PyJAGS format
+    """
+    if variable_names is None:
+        variable_names = tuple(pyjags_samples.keys())
+
+    assert isinstance(variable_names, tuple)
```

Comment:
It is not necessary. The code would work with an iterable. But it is more expressive. Since a tuple is immutable, It conveys the fact that the function does not change the input.